### PR TITLE
[admin-tool] Enable decompression in admin tool topic dumper

### PR DESCRIPTION
**Pull request by nisargthakkar**
## Enable decompression in admin tool topic dumper

This PR enables the `KafkaTopicDumper` in Admin tool to dump compressed records. It also fixes `TopicMessageFinder` to use serialized key without the chunking suffix to determine the partition

## How was this PR tested?

Tested manually.

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-admin-tool/src/main/java/com/linkedin/venice/TopicMessageFinder.java [hunk 1]
```diff
@@ -55,6 +55,11 @@ public static long find(
     int partitionCount = storeInfo.getPartitionCount();
     // Parse key string and figure out the right partition
     byte[] serializedKey = serializeKey(keyString, keySchemaStr);
+
+    // Partition assignment is always based on the non-chunked key.
+    int assignedPartition = new DefaultVenicePartitioner().getPartitionId(serializedKey, partitionCount);
+    LOGGER.info("Assigned partition: {} for key: {}", assignedPartition, keyString);
```
[the remark this answers is not in the document]
Comment (nisargthakkar) [reply clients/venice-admin-tool/src/main/java/com/linkedin/venice/TopicMessageFinder.java:61]: This is done only once in Admin tool `--query-kafka-topic` command unless someone is scripting around admin tool

[the remark this answers is not in the document]
Comment (nisargthakkar) [reply clients/venice-admin-tool/src/main/java/com/linkedin/venice/TopicMessageFinder.java:61]: This is done only once in Admin tool `--query-kafka-topic` command unless someone is scripting around admin tool

**Review by adamxchen — APPROVED** [review]
LGTM. Just want to confirm that you have tested it on both compressed store, and non-compressed store?

**Comment by nisargthakkar** [comment]
Yeah, I tested Gzip, zstd and no_op compression stores

**Review by adamxchen — APPROVED** [review]

**Pull request merged**